server/src/main/java/de/uftos/services/StudentGroupService.java

```diff
@@ -114,6 +114,10 @@ public StudentGroupResponseDto create(StudentGroupRequestDto rawGroup) {
       throw new ResponseStatusException(HttpStatus.BAD_REQUEST,
           "Could not find all grades by id");
     }
+    List<Grade> allGrades = this.gradeRepository.findAll();
```

Comment:
What is this loop for?
Shouldn't there be no grade containing a reference to the new student group?

Reply:
Yeah i don't know why i copied it over to create, it is obviously not needed.